firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,491 @@
+          Cu.import("resource://gre/modules/Services.jsm");
+
+          return  Services.vc.compare(arguments[0], arguments[1]);
+          """, script_args=[info['build_post']['version'], info['build_pre']['version']])
+
+        assert check >= 0, 'The version of the upgraded build is higher or equal'
+
+        # If there was an update, the post build id should be identical with the patch
+        if info['patch']['buildid']:
+            assert info['build_post']['buildid'] == info['patch']['buildid'], \
+                'The post buildid is equal to the buildid of the update'
+
+        # If a target build id has been given, check if it matches the updated build
+        info['target_buildid'] = update_data['update'].get('target_buildid')
+        if info['target_buildid']:
+            assert info['build_post']['buildid'] == info['target_buildid'], \
+                'Post buildid matches target buildid of the update patch'
+
+        # An upgrade should not change the builds locale
+        assert info['build_post']['locale'] == info['build_pre']['locale'],\
+            'The locale of the updated build has not been changed'
+
+        # Check that no application-wide add-ons have been disabled
+        assert info['build_post']['disabled_addons'] == info['build_pre']['disabled_addons'],\
+            'No application-wide add-ons have been disabled by the update'
+
+    def force_fallback(self):
+        """Update the update.status file and set the status to 'failed:6'"""
+        with open(os.path.join(self.staging_directory, 'update.status'), 'w') as f:
+            f.write('failed: 6\n')
+
+    def init_update_tests(self, fallback=False):
+        """Initialize all the data used by update tests.
+
+        :param fallback: It is a fallback update
+        """
+
+        # Prepare persisted object for update results
+        # If an update fails the post build will be the same as the pre build.
+        persisted = {
+            'update': {
+                'index': 0,
+                'staging_path': self.staging_directory},
+            # Create results object with information of the unmodified pre build
+            # TODO: Lets change to persisted.update.results once the issue is fixed:
+            # https://github.com/mozilla/mozmill-automation/issues/175
+            'updates': [
+                {
+                    'build_pre': self.build_info,
+                    'build_post': self.build_info,
+                    'fallback': fallback,
+                    'patch': {},
+                    'success': False}]}
+
+        # If requested modify the default update channel. It will be active
+        # after the next restart of the application
+        if persisted['update']['channel']:
+            # Backup the original content and the path of the channel-prefs.js file
+            persisted['update']['default_update_channel'] = {
+                'content': self.update_channel.file_contents,
```

Comment:
As we talked in the meeting yesterday please file a follow-up bug so we do copies of the files instead.